File: Arrays/ProblemSolutions/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i in range(0, len(nums)):
            if target - nums[i] not in hashmap:
                hashmap[nums[i]] = i
            else:
                return [hashmap[target-nums[i]], i]

# Question: https://leetcode.com/problems/two-sum/

File: HashMap/twoSum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int res[] = new int[2];
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(target-nums[i])){
                res[0] = map.get(target-nums[i]);
                res[1] = i;
            }
            map.put(nums[i],i);
        }
        return res;
    }
}

// Question: https://leetcode.com/problems/two-sum/?envType=study-plan-v2&envId=top-interview-150
